### DISCUSSION TITLE
Main Major Issues With the Bot

### DISCUSSION THREAD

**Discussion by VictimaVitae** [charlesbel/Microsoft-Rewards-Farmer#368] [General]
Here are the main issues with the bot, that ideally we should work on and implement countermeasures against, as soon as we can:

**1.** The bot has high power and data consumption (huge deal).

**2.** The bot is slow.

**3.** The United States (US) has significantly more daily points available, for anyone outside the US this is a lot of potential points missed. Could we integrate a VPN or other measure into the bot to allow people overseas to capitalise on these points? Some more info [here](https://www.reddit.com/r/MicrosoftRewards/comments/14rmswm/living_sticky_4_updated_daily_tutorial_faq_quests/).

**4.** The bot is not self-sufficient, when errors occur it must be manually restarted. This causes it to repeat actions it has already completed, this especially problematic when using multiple accounts, where many resource intensive actions must are repeated redundantly. A useful feature would be for the bot to remember accounts that it has completed, note the time they were completed and then not use them in future restarts during that day (by day I mean before 12 am, since Microsoft resets challenges by the day, rather than by using a timer).

**5.** The bot times out randomly during mobile searches and when it does it continues to do so indefinitely. This issue was fixed here by a legend: https://github.com/charlesbel/Microsoft-Rewards-Farmer/pull/276. Why hasn't this been integrated into the original script? Edit: this pull request was closed by the creator, make sure you use it!

**6.** The bot has gotten my accounts locked. Why haven't these enhancements been implemented: https://github.com/charlesbel/Microsoft-Rewards-Farmer/issues/282 and https://github.com/charlesbel/Microsoft-Rewards-Farmer/issues/223, to solve this issue.

**7.** Lastly, this is not so much a big problem, but the script is only usable by people with knowledge in coding. I have zero knowledge in coding, I somehow used my intuition to download this thing and get it running, along with Googling things. Maybe one day this may become an extension or application. As an incentive donations could be made in the application, to people who participated in the development of the bot.

**Comment by VictimaVitae** [comment]
Are there any other better and more updated bots out there?

**Comment by VictimaVitae** [comment]
For my number three issue I used Bing Precise mode to potentially solve the issue by prompting the AI to edit the code to implement [bobdabear](https://github.com/bobdabear)'s advice, see: https://github.com/charlesbel/Microsoft-Rewards-Farmer/issues/282. I harnessed Microsoft's AI to help me break Microsoft's terms of service 😂. Keep in mind I know nothing about coding.

Basically, I asked Bing to implement changes in the searches.py file because it seemed obvious that this was relevant since @bobdabear mentioned this in his enhancement about improving the search code. From this Bing suggested other files I could give it, so that it could find and change them. It seems to be working, although it still follows the order daily sets, then punch cards and then more promo.

**Comment by udu3324** [comment]
> Here are the main issues with the bot, that ideally we should work on and implement countermeasures against, as soon as we can:
> 
> 1. The bot has high power and data consumption (huge deal).
> 2. The bot is slow.
> 3. The bot has gotten my accounts locked. Why haven't these enhancements been implemented: [[ENHANCEMENT] Improved Random & Randomized order #282](https://github.com/charlesbel/Microsoft-Rewards-Farmer/issues/282) and [Add Account status reporting (Banned, Suspended, etc...) #223](https://github.com/charlesbel/Microsoft-Rewards-Farmer/issues/223), to solve this issue.
> 4. The bot times out randomly during mobile searches and when it does it continues to do so indefinitely. This issue was fixed here by a legend: [Mobile search fix, less CPU usage, & verbose notifs  #276](https://github.com/charlesbel/Microsoft-Rewards-Farmer/pull/276). Why hasn't this been integrated into the original script? No idea why this was closed by original creator.
> 5. Lastly, this is not so much a big problem, but the script is only usable by people with knowledge in coding. I have zero knowledge in coding, I somehow used my intuition to download this thing and get it running, along with Googling things. I can give people in the same boat instructions on how to get this thing running with if they need help.

msr bot is dead and there's no point in continuing

**Reply by VictimaVitae** [reply]
It may be dead, but not forgotten and it still works!

**Reply by udu3324** [reply]
sure, but now you have the problem of pointing out problems that no one has the time to solve

**Reply by VictimaVitae** [reply]
Fair enough I am not disagreeing, completely agree with you. Reckon this thing still has untapped potential though, that other people with time might tap into, and people have probably made projects branching from this already.

**Reply by bobdabear** [reply]
I'll take a closer look at everything this week... If I have the time I'll go ahead and code up the changes otherwise I'll start with just giving you more details on where the changes can be made and how so you have a bit more info to give Bing to work with.

**Reply by VictimaVitae** [reply]
@bobdabear your time and other people's time is precious, make sure you evaluate whether the return will be worth it. One day it may very well be if we create an application with donations, but I cannot foresee this in the near future. At least we will be stocked up on Bing points! 😂 I respect you!
